### PR TITLE
feat: add shields.io badge output format for CI/README status

### PR DESCRIPTION
**Pull request by iolairus**
## Summary

Add `--format badge` output format that generates JSON conforming to the [shields.io endpoint schema](https://shields.io/endpoint) for displaying scan status on READMEs or PR descriptions.

### Core Implementation
- **`borderlint/report.py`**: New `badge()` function generating shields.io endpoint JSON
  - Policy mode: `clean` (green), `{N} flagged` (red on failures, yellow on warnings)
  - Inventory mode: `{N} flows` (blue)
  - Non-gating export: always exits 0 regardless of violations

### CLI Integration
- **`borderlint/cli.py`**: Added `badge` to format choices and renderers dict

### Tests
- **`tests/test_borderlint.py`**: 4 new tests covering all badge scenarios
  - Policy mode clean (green)
  - Policy mode violations (red)
  - Inventory mode (blue)
  - Exit code verification (always 0)

### Documentation
- **`README.md`**: Added Badge section with usage examples and GitHub Actions workflow for publishing to shields.io

## Usage

```bash
borderlint scan . --policy residency.json --classification customer-pii --format badge
```

Output:
```json
{"schemaVersion": 1, "label": "borderlint", "message": "clean", "color": "green"}
```

## Traceability

- **Change**: add-badge-format
- **Spec**: openspec/changes/add-badge-format/specs/badge-output/spec.md
- **Tasks**: All 12 tasks completed (implementation, CLI, tests, documentation)

## Testing

All 131 tests pass, including 4 new badge-specific tests.

### CODE REVIEW

**Comment by iolairus** [comment]
Reviewed #67. The **badge** feature itself is clean and well-built: `report.badge()` is a tidy stdlib-only renderer, the color/message logic matches the spec, and the four tests (clean/violations/inventory/exit-code) cover the acceptance scenarios. No issues with the badge code.

**The blocking concern is structural, not code quality: this PR bundles three independent features into one branch.**

It contains, verbatim, the entire HTML report (#63) and the entire KB website (#64) — same `report.html`, same `scripts/kb_site.py`, same `.github/workflows/pages.yml`, same `openspec/changes/add-html-report-format/*` and `add-kb-website/*` directories. Those two PRs are already open and under review. Consequences:

1. **Guaranteed merge conflicts / double-application.** Merging #63 then #67 (or vice-versa) will conflict on `borderlint/cli.py` (the `--format` choices line, the renderers dict, the export tuple), `borderlint/report.py` (the `_arrangements`→`_arrangement_ids` refactor + `html()`), and `scripts/kb_site.py`. Worse, if both land, the `openspec/changes/` dirs get duplicated and archive-time validation will collide.
2. **Review and revert become all-or-nothing.** A defect in the badge can't be reverted without also ripping out the HTML report and KB site. The repo's own convention (per `AGENTS.md`) is *one concern per change, one change per branch* — this violates that.
3. **The `cli-and-reporting` delta in this PR also has the same MODIFIED/ADDED hygiene issue** I flagged on #63/#65: the `Scan command` requirement is unchanged text filed under `## MODIFIED Requirements`.

**Recommendation:** split this into the badge-only change. The cleanest path is to rebase `feature/badge` onto current `main` *after* #63 and #64 merge, dropping the `html`/`kb_site`/`pages.yml` hunks and their `openspec/changes/add-html-report-format` + `add-kb-website` directories (those belong to #63/#64). What should remain: `borderlint/report.py` `badge()`, the `badge` wiring in `cli.py`, the `badge-output` spec + the `badge` line in `cli-and-reporting` delta, the badge tests, and the README "Badge" section. The badge feature is worth shipping — just not bundled with two other PRs' work.

If the intent was deliberately to fast-track everything together, that's a process call for the maintainer, but as written this PR can't be merged alongside #63/#64 without conflicts. Happy to help split it if useful.

**Comment by iolairus** [comment]
Follow-up / clarification: the maintainer confirms this PR is the **canonical combined change** — the HTML report, KB site, and badge are intended to land together as one sequential merge, not as three independent PRs. In that light the bundling is intentional and the "split it" recommendation above is withdrawn.

The only remaining action to keep the merge clean: since #63 and #64 are the same work split out as standalone PRs, they should be **closed as superseded by this PR** rather than merged, so they aren't applied on top of #67 (which would duplicate `cli.py`/`report.py`/`kb_site.py` and the `openspec/changes/` dirs). With #63/#64 closed, #67 merges cleanly as the single source of truth.

Badge code itself remains good — no changes needed there.

**Comment by iolairus** [comment]
Correction to my follow-up: the intent is **merged in sequence**, not superseded. #63 (HTML report) and #64 (KB site) are real, separate PRs that merge first; #67 then adds the badge on top of the already-merged HTML report + KB site.

Practical consequence for this branch: once #63 and #64 are merged to `main`, `feature/badge` should be **rebased onto `main`**. After that rebase the html/`kb_site`/`pages.yml` hunks and the `openspec/changes/add-html-report-format` + `add-kb-website` directories will already be in `main` and drop out of this PR's diff, leaving #67 as a clean badge-only change. No need to close #63/#64 — they merge normally, in order, ahead of this one.

Badge code itself remains good — no changes needed there.

**Review by iolairus — COMMENTED** [review]
Badge review — the renderer itself is clean (correct shields schema, non-gating export semantics match house rules, tests follow the SARIF pattern). Four findings, plus one merge-order note.

1. **[blocking] cli-and-reporting delta drops PR #63's `html` edits** — inline comment on the delta spec.
2. **Yellow warn state is implemented and documented but never specced** — inline comment on `report.py`.
3. **Inventory zero-flow divergence**: `badge()` emits `"clean"`/blue when 0 flows, but the badge-output spec's message requirement says `"{N} flows"` in inventory mode. Either emit `"0 flows"` or add the clean case to the spec.
4. **Docs nits**: the README "Output & CI" enumeration gained HTML but not badge; CAPABILITIES.md has no badge row (D8 would match the D-table pattern — note D7/HTML arrives via #63's branch which this PR is stacked on).

**Merge order**: this branch contains `feature/add-html-report-format` and `feature/add-kb-website` merged in, so the PR diff currently bundles #63 + #64. That's fine *only if* #63 and #64 merge first (the diff auto-shrinks to badge-only). Merging #67 first would land both of those without their own review closing out. Suggest holding #67 until #63/#64 are in.

File: openspec/changes/archive/2026-07-13-add-badge-format/specs/cli-and-reporting/spec.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+## MODIFIED Requirements
+
+### Requirement: Scan command
```
Comment (iolairus) [inline openspec/changes/archive/2026-07-13-add-badge-format/specs/cli-and-reporting/spec.md:3]: **Blocking (archive-order hazard).** Both MODIFIED requirements here were copied from the *old* main-spec text, not the text as amended by PR #63 — which this branch's *code* is stacked on. "Scan command" enumerates `…SBOM, evidence, or shields.io badge endpoint` with **no HTML**, and "CI exit code" lists `sbom`/`evidence`/`badge` with **no `html`** and drops the "HTML export does not gate" scenario. If this change archives after #63, these copies overwrite the main spec and silently revert #63's html edits — the exact hazard of MODIFIED-with-stale-content.

Fix: copy the requirement text from `openspec/changes/add-html-report-format/specs/cli-and-reporting/spec.md` (which includes html in both places, plus its scenario) and add `badge` on top: enumeration `…SBOM, evidence, HTML, or shields.io badge endpoint`; exception clause `(--format sbom, --format evidence, --format html, or --format badge)`; keep both the HTML and badge non-gating scenarios.

File: borderlint/report.py [hunk 2]
```diff
@@ -482,3 +599,38 @@ def evidence(findings, kb, policy=None, envelope=None) -> str:
...
+    if policy is None:
+        # Inventory mode: blue badge with flow count
+        count = len(findings)
+        message = f"{count} flows" if count > 0 else "clean"
+        color = "blue"
+    else:
+        # Policy mode: check for failures
+        has_failures = any(f.severity == "fail" for f in findings)
+        flagged = sum(1 for f in findings if f.severity in ("fail", "warn"))
+        if has_failures:
```
Comment (iolairus) [inline borderlint/report.py:621]: **Yellow is implemented but not specced.** This warn-only branch (`"{N} flagged"`, `yellow`) and its README documentation don't exist in the badge-output spec: the color requirement says green (clean) / red (violations) / blue (inventory) only, and the message requirement says the message SHALL be `"clean"` when no violations exist — a warn-only scan arguably has no violations, so spec and code disagree on both color and message. Design decision 1 also omits yellow.

Per the repo rule ("if during implementation you discover the spec was wrong or incomplete, stop, update the spec, then continue"): either add the yellow state to the badge-output spec (color + message requirements, one scenario, e.g. WHEN a scan yields warnings and no failures THEN message `"{N} flagged"` and color `"yellow"`) and to design decision 1 — or drop the yellow branch and README mention. There's also no test pinning the yellow path either way.

File: README.md [hunk 3]
```diff
@@ -147,7 +150,7 @@ APP 8) as reference links. (`home_regime` `pdpo`/`pipl` is still accepted.)
   → PDPO / Macao PDPA / PIPL + the matching GBA Standard Contract; **APAC/EMEA seats `jp` (APPI), `kr`
   (PIPA), `sg`/`my` (PDPA s.26 / s.129), `au` (APP 8), `uk` (UK IDTA), `eu` (GDPR)** → their transfer
   mechanism. PIPL cross-border and GDPR are also surfaced for those destinations.
-- **Output & CI:** text / JSON / Mermaid / SARIF / SBOM, an SBOM **`diff`** gate for new
+- **Output & CI:** text / JSON / Mermaid / SARIF / SBOM / HTML, an SBOM **`diff`** gate for new
```
Comment (iolairus) [inline README.md]: This enumeration gained HTML (from the #63 branch) but not badge — `text / JSON / Mermaid / SARIF / SBOM / HTML / badge` for consistency with the `--format` line above.

**Comment by iolairus** [comment]
Rebased per the agreed sequence: #63 and #64 are now merged to `main` (squashed as 9fda2cb and 9049a01), and `feature/badge` has been rebuilt as `main` + the badge commit only (`4e39fb7`, cherry-picked from `5cef26e`, applied clean, 131 tests green). The PR diff is now badge-only.

Still open from the review before this can merge: (1) the `cli-and-reporting` delta must copy the requirement text **as now amended in `openspec/specs/cli-and-reporting/spec.md`** (html is in the main spec since #63 archived — add `badge` on top of that text, don't drop html); (2) the yellow warn-state needs spec/design/test coverage or removal; (3) inventory zero-flow `"clean"` vs spec'd `"{N} flows"`; (4) README/CAPABILITIES enumeration nits.

**Comment by iolairus** [comment]
Re-checked after the fix commits. Four of five findings are properly addressed — Scan-command delta copy now current (HTML + badge), yellow specced with a warnings-only scenario, `"0 flows"` specced and implemented, README enumeration + CAPABILITIES D8 in. Three items before this can merge:

1. **[still blocking] The "CI exit code" delta copy remains stale.** Its exception clause reads `(--format sbom, --format evidence, or --format badge)` — missing `--format html` — and the "HTML export does not gate" scenario was dropped. The current main spec (post-#63 archive) reads `(--format sbom, --format evidence, or --format html)` and includes that scenario; archiving this delta as-is would strip html's export exception from `openspec/specs/cli-and-reporting/spec.md`. Fix exactly like the Scan-command requirement: copy the current main-spec text verbatim, add `--format badge` to the clause, keep both the HTML and badge non-gating scenarios. (The *code* is already correct — `("sbom", "evidence", "html", "badge")` — only the spec copy is wrong.)

2. **Missing tests for the two new scenarios.** "Policy warnings only yield yellow badge" and "0 flows when zero flows are detected" now exist as acceptance scenarios but have no tests — the four badge tests cover clean/red/inventory/exit-code only. Repo convention is a test per scenario; both are 5-liners with the existing `_pol`/`evaluate` helpers.

3. **Rebase needed again**: main advanced with #68 (Java/Kotlin), and the branch has gone CONFLICTING — most likely the CAPABILITIES D8 row and README enumeration colliding with #68's A8 row and languages edits. `git fetch && git rebase origin/main` should be trivial (keep both rows, both enumerations).
